lib/matplotlib/backend_bases.py

```diff
@@ -3103,34 +3098,30 @@ def press_pan(self, event):
 
     def drag_pan(self, event):
         """Callback for dragging in pan/zoom mode."""
-        for a, ind in self._xypress:
-            #safer to use the recorded button at the press than current button:
-            #multiple button can get pressed during motion...
-            a.drag_pan(self._button_pressed, event.key, event.x, event.y)
+        for ax in self._pan_info["axes"]:
+            # Using the recorded button at the press is safer than the current
+            # button, as multiple buttons can get pressed during motion.
+            ax.drag_pan(self._pan_info["button"], event.key, event.x, event.y)
         self.canvas.draw_idle()
 
     def release_pan(self, event):
         """Callback for mouse button release in pan/zoom mode."""
-
-        if self._button_pressed is None:
+        if self._pan_info is None:
             return
-        self.canvas.mpl_disconnect(self._id_drag)
+        self.canvas.mpl_disconnect(self._pan_info["cid"])
         self._id_drag = self.canvas.mpl_connect(
             'motion_notify_event', self.mouse_move)
-        for a, ind in self._xypress:
-            a.end_pan()
-        if not self._xypress:
-            return
-        self._xypress = []
-        self._button_pressed = None
+        for ax in self._pan_info["axes"]:
+            ax.end_pan()
+        self._draw()
```

Comment:
Could it have an unwanted effect that you move `_draw()` above `release()`?

Reply:
Likely not, but to be safe I moved the whole block *below* release() (still making the order consistent with release_zoom); now it's just push_current() which goes below release() which should be even safer (in theory you can further change the view in an overridden release() but I'm not too worried about that and that's deprecated anyways...)